.travis.yml

```diff
@@ -36,6 +39,49 @@ matrix:
       env: PYTHON_VERSION=3.6 PIP_DEPENDENCIES='git+https://github.com/sphinx-doc/sphinx.git#egg=sphinx coveralls'
            CONDA_DEPENDENCIES="setuptools cython numpy pytest-cov"
 
+    # Test gcc on OSX
+    - os: osx
+      env:
+        - PYTHON_VERSION=3.5
+      before_install:
+        - rm /usr/local/include/c++
+        - brew install gcc6
+        - CC=gcc-6
+        - CXX=g++-6;
+
+    # Test manual install of LLVM's clang and LLVM-OpenMP
+    - os: osx
+      env:
+        - PYTHON_VERSION=3.5
+        - CONDA_DEPENDENCIES="setuptools sphinx cython numpy pytest-cov cmake"
+      before_install:
+        - wget http://releases.llvm.org/6.0.0/clang+llvm-6.0.0-x86_64-apple-darwin.tar.xz
+        - tar -xvf clang+llvm-6.0.0-x86_64-apple-darwin.tar.xz
+        - loc=`pwd`
+        - path=$loc"/clang+llvm-6.0.0-x86_64-apple-darwin"
+        - export PATH=$path"/bin":$PATH
+        - export C_INCLUDE_PATH=$path"/include":$C_INCLUDE_PATH
+        - export LD_LIBRARY_PATH=$path"/lib":$LD_LIBRARY_PATH
+        - export CC=$path"/bin/clang"
+        - svn co http://llvm.org/svn/llvm-project/openmp/trunk openmp
+        - mkdir openmp-build && cd openmp-build
+        - cmake ../openmp
+        - make
+        - make install
+        - cd ../
+
+    # Test without OpenMP support
+    # Since this is using the OS shipped version of clang, this test doubles up
+    # as an exploritory test for when the shipped version has automatic OpenMP support.
+    # This test will then fail and at such a time should not be reomved but
+    # altered to explicitly remove OpenMP support.
+    - os: osx
+      env:
+        - PYTHON_VERSION=3.5
+        - export TEST_OPENMP="False"
+      before_install:
+        - echo "Default before_install ignored"
```

Comment:
Rather than overriding before_install here to disable it, what about swapping this build with the one that has the default ``before_install``? Then in the default ``before_install``, just set TEST_OPENMP to False if the platform is OSX.

Reply:
The idea is to test both `TEST_OPENMP` states but since there are two default OSX builds I can highjack one if you're cool with that?  I'd prefer not to though, tests shouldn't really overlap. The two counterparts are:
```python
env:
  matrix:
    - PYTHON_VERSION=3.5

- os: osx
      env:
        - PYTHON_VERSION=3.5
        - export TEST_OPENMP="False"
      before_install:
        - echo "Default before_install ignored"
```

Reply:
I guess what I meant is that the original OSX test was the one without OpenMP anyway, so why not make the three new ones the ones that test different OpenMP enabled compilers, and keep the default build as the one with the default compilers?